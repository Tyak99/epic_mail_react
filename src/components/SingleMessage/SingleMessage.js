import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { getSingleMessage } from '../../../store/actions/messageAction';
import Navigation from '../Navigation/Navigation';
import styles from './singleMessage.css';

export class SingleMessage extends React.Component {
  componentDidMount() {
    const { match, getMessage } = this.props;
    getMessage(match.params.id);
  }

  render() {
    const { message, error } = this.props;
    let display = 'Loading...';
    if (error) {
      display = (
        <h2>
          {error}
        </h2>
      );
    }
    if (message) {
      display = (
        <Fragment>
          <div className={styles.threadGroup}>
            <div className={styles.subject}>
              <h2>
                {message.subject}
              </h2>
            </div>
            <div className={styles.thread}>

              <div className={styles.subjectItem}>
                <p className={styles.from}>
                  <span style={{ marginRight: '10px' }}>
                    From
                  </span>
                  <em className={styles.sender}>
                    {message.senderid}
                  </em>
                </p>
                <p className={styles.to}>
                  <span style={{ marginRight: '10px' }}>

                    To:
                  </span>
                  <em className={styles.receiver}>
                    {message.receiverid}
                  </em>
                </p>
                <div className={styles.emailDate}>
                  <p style={{ color: 'grey' }}>
                    {message.created_at}
                  </p>
                </div>
                <div className={styles.icons}>
                  <a href="./compose.html">
                    <span className={styles.reply}>
                      <FontAwesomeIcon icon="reply" />
                    Reply
                    </span>
                  </a>
                  <span className={styles.btnDelete}>
                    <FontAwesomeIcon icon="trash-alt" />
                    Delete
                  </span>
                </div>
              </div>

            </div>
            <div className={styles.threadBody}>
              {message.message}
            </div>
          </div>
        </Fragment>
      );
    }
    return (
      <div>
        <Navigation />
        {display}
      </div>
    );
  }
}

// export default SingleMessage;
SingleMessage.defaultProps = {
  message: {},
  error: null,
};
SingleMessage.propTypes = {
  message: PropTypes.shape({
    subject: PropTypes.string,
    message: PropTypes.string,
  }),
  getMessage: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.object,
  }).isRequired,
  error: PropTypes.string,
};

const mapStateToProps = state => ({
  message: state.messages.message,
  error: state.messages.error,
});
const mapDispatchToprops = dispatch => ({
  getMessage: id => dispatch(getSingleMessage(id)),
});
export default connect(mapStateToProps, mapDispatchToprops)(SingleMessage);
