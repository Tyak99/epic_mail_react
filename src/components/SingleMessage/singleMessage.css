.threadGroup {
    height: 90vh;
    overflow: scroll;
    padding: 0 30px;
  }


.threadBody {
    margin-top: 20px;
    line-height: 1.7rem;
  }
  
  .from .to {
      margin-right: 10px
  }
  .sender {
    color: darkorchid;
    margin-right: 5px;
  }
  
  .receiver {
    color: firebrick;
  }
  
  .icons a {
    text-decoration: none;
  }
  
  .icons .reply,
  .btnDelete {
    padding: 6px;
    margin-right: 10px;
    border: 2px solid #a8c9da;
    cursor: pointer;
    color: darkred;
  }
  
  .icons .reply:hover,
  .btnDelete:hover {
    background-color: #a8c9da;
  }
  
  hr {
    height: 10px;
    border: 0;
    box-shadow: 0 10px 10px -10px #8c8b8b inset;
  }
  