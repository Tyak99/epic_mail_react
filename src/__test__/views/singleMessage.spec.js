import React from 'react';
import { shallow } from 'enzyme';
import { SingleMessage } from '../../components/SingleMessage/SingleMessage';

const props = {
  message: {},
  match: { params: {} },
  getMessage: () => {},
};
describe('component: SingleMessage', () => {
  it('should render correctly', (done) => {
    const wrapper = shallow(<SingleMessage {...props} />);
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
