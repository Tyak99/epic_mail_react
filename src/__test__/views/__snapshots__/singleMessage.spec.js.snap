// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: SingleMessage should render correctly 1`] = `ShallowWrapper {}`;
